bridges/ethMultiversX/bridgeExecutor_test.go

```diff
@@ -401,96 +396,92 @@ func TestEthToMultiversXBridgeExecutor_GetAndStoreBatchFromEthereum(t *testing.T
-			expectedDepositData := codec.EncodeCallData(emtpyCallData)
-			assert.Equal(t, string(expectedDepositData), string(executor.batch.Deposits[0].Data))
-		})
-		t.Run("correct data", func(t *testing.T) {
-			t.Parallel()
-
-			args := createMockExecutorArgs()
-			providedNonce := uint64(8346)
-			depositNonce := uint64(100)
-			depositData := string(codec.EncodeCallData(parsers.CallData{
-				Type:      parsers.DataPresentProtocolMarker,
-				Function:  "function",
-				GasLimit:  37,
-				Arguments: []string{"arg1", "arg2"},
-			}))
-			expectedBatch := &bridgeCommon.TransferBatch{
-				ID: providedNonce,
-				Deposits: []*bridgeCommon.DepositTransfer{
-					{
-						Nonce: depositNonce,
-					},
-				},
-			}
-			args.EthereumClient = &bridgeTests.EthereumClientStub{
-				GetBatchCalled: func(ctx context.Context, nonce uint64) (*bridgeCommon.TransferBatch, bool, error) {
-					assert.Equal(t, providedNonce, nonce)
-					return expectedBatch, true, nil
+		t.Parallel()
+
+		args := createMockExecutorArgs()
+		providedNonce := uint64(8346)
+		depositNonce := uint64(100)
+		depositData := "testData"
+		expectedBatch := &bridgeCore.TransferBatch{
+			ID: providedNonce,
+			Deposits: []*bridgeCore.DepositTransfer{
+				{
+					Nonce: depositNonce,
 				},
-				GetBatchSCMetadataCalled: func(ctx context.Context, nonce uint64, blockNumber int64) ([]*contract.ERC20SafeERC20SCDeposit, error) {
-					return []*contract.ERC20SafeERC20SCDeposit{{
-						DepositNonce: big.NewInt(0).SetUint64(depositNonce),
-						CallData:     depositData,
-					}}, nil
+			},
+		}
+		args.EthereumClient = &bridgeTests.EthereumClientStub{
+			GetBatchCalled: func(ctx context.Context, nonce uint64) (*bridgeCore.TransferBatch, bool, error) {
+				assert.Equal(t, providedNonce, nonce)
+				return expectedBatch, true, nil
+			},
+			GetBatchSCMetadataCalled: func(ctx context.Context, nonce uint64, blockNumber int64) ([]*contract.ERC20SafeERC20SCDeposit, error) {
+				return []*contract.ERC20SafeERC20SCDeposit{{
+					DepositNonce: big.NewInt(0).SetUint64(depositNonce),
+					CallData:     depositData,
+				}}, nil
+			},
+		}
+		executor, _ := NewBridgeExecutor(args)
+		err := executor.GetAndStoreBatchFromEthereum(context.Background(), providedNonce)
+
+		assert.Nil(t, err)
+		assert.True(t, expectedBatch == executor.GetStoredBatch()) // pointer testing
+		expectedDepositData := []byte{bridgeCore.DataPresentProtocolMarker, 0, 0, 0, byte(len(depositData))}
+		expectedDepositData = append(expectedDepositData, []byte(depositData)...)
+		assert.Equal(t, string(expectedDepositData), string(executor.batch.Deposits[0].Data))
+	})
+	t.Run("should add deposits metadata for sc calls with a data starting with missing data marker", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockExecutorArgs()
+		providedNonce := uint64(8346)
+		depositNonce := uint64(100)
+		depositData := string([]byte{bridgeCore.MissingDataProtocolMarker}) + "testData"
```

Comment:
why do you also add the MissingDataProtocolMarker, this should not be added when calling the deposit function on solidity, only the actual data.

Reply:
just for test purposes, this will emphasize the fact that anything except the missing marker will be wrapped. on L466 (same test) we have this code:
```GO
   expectedDepositData := []byte{bridgeCore.DataPresentProtocolMarker, 0, 0, 0, byte(len(depositData))}
   expectedDepositData = append(expectedDepositData, []byte(depositData)...)
```
the `expectedDepositData` is `0x01 0x00 0x00 0x00 0x09 01 't' 'e' 's' 't' 'D' 'a' 't' 'a'`